geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -42,16 +45,65 @@ public RedisList() {
     this.elementList = new SizeableByteArrayList();
   }
 
+  /**
+   * @param start start index of desired elments
+   * @param stop stop index of desired elments
+   * @return list of elements in the range (inclusive).
+   */
+  public List<byte[]> lrange(int start, int stop) {
+    start = transformNegIndexToPosIndex(start);
+    stop = transformNegIndexToPosIndex(stop);
+
+    // Out of range negative index changes to 0
+    start = Math.max(0, start);
+
+    int elementSize = elementList.size();
+    if (start > stop || elementSize <= start) {
+      return Collections.emptyList();
+    }
+
+    // Out of range positive stop index changes to last index available
+    stop = elementSize <= stop ? elementSize - 1 : stop;
+
+    List<byte[]> result = new LinkedList<>();
+    int curIndex;
+
+    // Finds the shortest distance to access nodes in range
+    if (start <= elementSize - stop - 1) {
+      // Starts at head to access nodes at start index then iterates forwards
+      curIndex = 0;
+      ListIterator<byte[]> iterator = elementList.listIterator(curIndex);
+
+      while (iterator.hasNext() && curIndex <= stop) {
+        byte[] element = iterator.next();
+        if (start <= curIndex) {
+          result.add(element);
+        }
+        curIndex = iterator.nextIndex();
+      }
+    } else {
+      // Starts at tail to access nodes at stop index then iterates backwards
+      curIndex = elementSize - 1;
+      ListIterator<byte[]> iterator = elementList.listIterator(elementSize);
+
+      while (iterator.hasPrevious() && start <= curIndex) {
+        byte[] element = iterator.previous();
```

Comment:
It looks like "element" is only used inside the "if" so should this call be moved inside the if?